OpenRA.Mods.Common/Traits/RevealOnFire.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Linq;
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Reveal this actor to the target's owner when attacking.")]
+	public class RevealOnFireInfo : ConditionalTraitInfo
+	{
+		[Desc("The armament types which trigger revealing.")]
+		public readonly string[] ArmamentNames = { "primary", "secondary" };
+
+		[Desc("Duration of the reveal.")]
+		public readonly int Duration = 25;
+
+		[Desc("Radius of the reveal around this actor.")]
+		public readonly WDist Radius = new WDist(1536);
+
+		[Desc("Can this actor be revealed through shroud generated by the GeneratesShroud trait?")]
+		public readonly bool RevealedThroughGeneratedShroud = true;
```

Comment:
Nit: I'd change this from past to present tense (`RevealThrough...`). Makes it a tiny bit shorter, too.